src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -28,6 +31,29 @@ internal class BulkUpdateSaveResult : BaseSaveResult
         /// </summary>
         private readonly BulkUpdateGraph bulkUpdateGraph;
 
+        /// <summary> Link descriptors for some descriptors.</summary>
+        private readonly Dictionary<Descriptor, LinkDescriptor> linkDescriptors;
+
+        /// <summary>The bulk update response cache.</summary>
+        private CachedResponse cachedResponse;
+
+        /// <summary>The descriptor associated with a materializer.</summary>
+        private readonly Dictionary<Descriptor, IMaterializerState> materializerStateForDescriptor;
+
+        /// <summary>A list of all the operation responses for a bulk update response.</summary>
+        private readonly List<OperationResponse> operationResponses;
+
+        /// <summary>Holds the list of descriptors related to a visited descriptor that was added to the visited set.</summary>
+        private Stack<(int,IReadOnlyList<Descriptor>)> adjacentDescriptors;

```

Comment:
Given that this stack is used in a very limited context (only a single method), and its definition is a bit convoluted, I think it would be helfpul to make the summary more detailed, explaining how its used and structured and why it's there.

Reply:
> I'm suprised this tuple literal syntax compiles on all our supported frameworks.

@habbes It works because a reference to `System.ValueTuple` package was added to the project